pandas/core/indexing.py

```diff
@@ -1724,7 +1724,12 @@ def _setitem_with_indexer_frame_value(self, indexer, value: "DataFrame"):
 
         unique_cols = value.columns.is_unique
 
-        if not unique_cols and value.columns.equals(self.obj.columns):
+        if name == "iloc":
+            for i, loc in enumerate(ilocs):
```

Comment:
can you add a comment pointing back to this PR explaining why we're doing things differently for iloc vs loc

Reply:
Done